modules/admission-webhook-create.adoc

```diff
@@ -0,0 +1,328 @@
+// Module included in the following assemblies:
+//
+// * architecture/admission-controllers.adoc
+
+[id="admission-webhook-create_{context}"]
+= Creating an admission webhook
+
+.Procedure
+
+This procedure outlines high level steps to deploy a webhook server as a Daemonset as well as to deploy the webhook server as an aggregated API server. Deploying the webhook server as an aggregated API server allows other {product-title} components to communicate with the webhook using internal credentials and facilitates testing using the `oc` command. This also enables role based access control (RBAC) into the webhook, as well as preventing token information from other API servers being disclosed to the webhook. The API server calls `kube-apiserver` with its fully authenticated service account. The aggregated API server then calls to the webhook using the request header certificate. The webhook verifies incoming client certificates through the published request header aggregation CA.
+
+[NOTE]
+====
+The link:https://github.com/openshift/kubernetes-namespace-reservation[kubernetes-namespace-reservation] repository provides a full end-to-end example of a webhook server/aggregated API server deployment. Note that this repository has not been fully tested or validated by Red Hat.
+====
+
+. Build a webhook server container image and make it available to the cluster via an image registry.
+
+. Create a local CA key and certificate and use them to sign the webhook server's certificate signing request (CSR).
+
+. Create a YAML based definition of RBAC rules for the aggregated API service. Replace values within angle brackets `<>` accordingly:
```

Comment:
Though, I'm looking at this, and I don't see anywhere else in the docs where we generically say to replace everything in angle brackets. Plus this statement is only for this step, whereas they're going to need to do it throughout the example YAML files in subsequent steps.

I know we've been going back and forth with changing this from an end-to-end example to a more generic example, but I think that there are just too many replaceable values used throughout these YAML files. With how large the YAML examples are, I think that you'd need to do a callout for each place that they'd need to replace a value.

Also, you can still have this be a generic example, but use some test values, to reduce the number of replaceables a user needs to act upon to get the example to work. For example, pick a namespace (let's say `my-webhook-namespace` for example), and then actually create that, and use that throughout your examples. And then in the step where you create it, you could just give them a heads up that this example procedure uses the `my-webhook-namespace` or whatever, so they'll know to replace it if they want to call it something different. That alone would get rid of 15 replaceable values that you have. Same thing for names for just about anything, like the cluster roles/role bindings that are created. It's okay to use arbitrary names, because the names aren't really a critical setting.

However, I would be interested to see if anyone else on the @openshift/team-documentation has any recommendations on how to handle this magnitude of example YAML and replaceables.

Reply:
Thank you for your guidance on this point. I have removed the statement as well as the replaceable entries. I have now used examples and called each one out. I think it improves the procedure.